RyuSocks.Test/Auth/UsernameAndPasswordTests.cs

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) RyuSOCKS
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+using RyuSocks.Auth;
+using RyuSocks.Auth.Packets;
+using System;
+using System.Collections.Generic;
+using System.Security.Authentication;
+using Xunit;
+
+namespace RyuSocks.Test.Auth
+{
+    public class UsernameAndPasswordTests
+    {
+        private static readonly UsernameAndPasswordResponse _expectedUsernameAndPasswordResponse = new()
+        {
+            Version = AuthConsts.UsernameAndPasswordVersion,
+            Status = 0,
+        };
+
+        [Theory]
+        [InlineData("Username", "Password", true)]
+        [InlineData("InvalidUsername", "InvalidPassword", false)]
+        public void Authenticate_WorksOnRegisteredUserAuthenticationClientSide(string username, string password,
+            bool acceptable)
+        {
+            UsernameAndPasswordRequest expectedUsernameAndPasswordRequest =
+                new(username, password);
+            UsernameAndPasswordResponse expectedUsernameAndPasswordResponse = acceptable ? new UsernameAndPasswordResponse([AuthConsts.UsernameAndPasswordVersion, 0]) : new UsernameAndPasswordResponse([AuthConsts.UsernameAndPasswordVersion, 1]);
+
+            UsernameAndPassword usernameAndPassword = new()
+            {
+                Database = new Dictionary<string, string> { { "Username", "Password" } },
+                Username = username,
+                Password = password,
+                IsClient = true
+            };
```

Comment:
Since you are using `UsernameAndPassword` in most of your tests, I'd suggest making this a local field and implementing `IDisposible` for this class.

See: https://xunit.net/docs/shared-context#constructor

Reply:
can't do that
the site has X as icon and that reminds me of X(Twitter), literally impossible